decidim-core/lib/decidim/scopable.rb

```diff
@@ -68,10 +70,18 @@ def previous_scope
 
     private
 
+    # validation method for participatory spaces
     def scope_belongs_to_organization
       return if !scope || !organization
 
       errors.add(:scope, :invalid) unless organization.scopes.where(id: scope.id).exists?
     end
+
+    # validation method for components
+    def scope_belongs_to_participatory_space
+      return if !scopes_enabled? || !participatory_space
+
+      errors.add(:scope, :invalid) if participatory_space.out_of_scope?(scope)
+    end
```

Comment:
shouldn't we have this methods in the corresponding `ScopableComponent` and `ScopableParticipatorySpace` concerns?

Reply:
`validate :scope_belongs_to_organization` ensures that the `scopable` (space, component or any other resource) is scoped within the organization's scope, I think this is a good place for this method.

and the `:scope_belongs_to_participatory_space` method ensures that any resource scope (such as a component) is scoped within the spaces' scope, so yes I'll move this method to the `ScopableComponent` module where `validate :scope_belongs_to_participatory_space` is called.